Use MessageContainer for logging errors (#1155)
mono/linker: docs/error-codes.md

- There was an error writing the linked assembly 'output'. An exception with more details is printed.

#### `IL1012`: IL Linker has encountered an unexpected error. Please report the issue at https://github.com/mono/linker/issues

> - There was an unexpected error while linking. An exception with more details is printed.

Reviewer: Would be nice to add more info where to get the log from or what kind of details to log

#### `IL1012`: IL Linker has encountered an unexpected error. Please report the issue at https://github.com/mono/linker/issues

> - There was an unexpected error while linking. An exception with more details is printed to the MSBuild log, please share the stack trace seen in there with the IL Linker team to further investigate the cause and possible solution.

Reviewer 2: ```suggestion
- There was an unexpected error while linking. An exception with more details is printed to the MSBuild log. Please share this stack trace with the IL Linker team to further investigate the cause and possible solution.
```
